cmd/juju/caas/add.go

```diff
@@ -128,6 +119,21 @@ func (c *AddCAASCommand) newAPIRoot() (api.Connection, error) {
 	return c.NewAPIRoot()
 }
 
+func getKubeConfigSource(ctx *cmd.Context) (r io.Reader, err error) {
+	if ctx.Stdin != nil {
+		// Try to read from Stdin.
+		r = ctx.Stdin
+	} else {
+		// Try to read from kubeconfig file because no Stdin.
+		filename := clientconfig.GetKubeConfigPath()
+		r, err = os.Open(filename)
+		if err != nil {
+			return nil, errors.Trace(errors.Annotatef(err, "failed to read kubernetes config from '%s'", filename))
```

Comment:
Check for file not found and return a NotFoundError if required.
Then this block can be extracted as a helper function also called by DetectCredentials.